2863 django11 upgrade

sblack-usu opened the pull request:
<!--

Please read, and add your text at the bottom of this message.

Thanks for contributing code to HydroShare. In order to maintain code quality and expedite this process, please assist the development team by making sure the following is present in this pull request.

For more information, see https://docs.google.com/document/d/1dzxqlZW5fKNEyQSeKiSFq-SmS-VOPCva95XXkBjPExs

-->

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

<!-- Enter steps that a QA engineer, stakeholder, or user documentation writer would follow to test the positive or "successful" case of the functionality your code provides or fixes -->

- [x] Automated Testing

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Code coverage must not decrease so new functionality or code paths added during a bug fix must have appropriate tests written. Every test must pass, including PEP8 code formatting tests. -->

- [x] Sufficient User and Developer Documentation

<!-- Please email your positive test case lbrazil@cuahsi.org, who will make the decision regarding user documentation. -->

- [x] Passing Jenkins Build

<!-- Our Jenkins Instance is set up to automatically test every commit from a pull request. Every test must pass, including PEP8 code formatting tests. -->

- [x] Peer Code review and approval

<!-- This is the process by which a peer developer on the HydroShare team will read the changeset, provide feedback, and ultimately give a formal approval to the code before it passes PR status. -->

### Positive Test Case
1. Run it and poke around... this is not a feature, but an upgrade.

sblack-usu commented:
retest this, please

sblack-usu commented:
retest this, please

sblack-usu commented:
retest this, please

hyi commented on Dockerfile:
```diff
@@ -1,21 +1,9 @@
-### End - HydroShare Development Image Additions ###
+# inplaceedit in pip doesn't seem compatible with Django 1.11 yet...
+RUN pip uninstall -y django-inplaceedit
+RUN pip install git+https://github.com/theromis/django-inplaceedit.git@e6fa12355defedf769a5f06edc8fc079a6e982ec
```
@sblack-usu Could you update the base image to remove django-inplaceedit so it does not have to be uninstalled here as an extra step? Also my impression is that we are not using django-inplaceedit in HydroShare, so this might be a good chance to test it out and if we are not using it, we don't have to handle this upgrade incompatibility here.

sblack-usu replied:
I'm building hs_dockerbase without django-inplaceedit, locally now.  If we don't end up needing it, I'll remove both of these lines.

sblack-usu replied:
we need inplaceedit, on your suggestion, I removed it from hs_dockerbase so we don't have to uninstall it here before installing a different version.

sblack-usu replied:
looks like another package has django-inplaceedit as a dependency so it still needs to be reinstalled.

hyi commented on hs_modelinstance/views.py.old:
```diff
@@ -4,7 +4,7 @@ from django.views import generic
 from django.core.urlresolvers import reverse
 from django.contrib.auth.decorators import login_required
-from django.shortcuts import render_to_response
+#from django.shortcuts import render_to_response
```
@sblack-usu just remove this import rather than comment it out?

sblack-usu replied:
do you know what this file is for?  I really should be able to just remove the whole file (.py.old)

hyi replied:
I don't know what is file is for, but I am very sure this whole file should be removed, so just go ahead to remove the whole file.

sblack-usu replied:
removed

hyi commented on hsctl:
```diff
@@ -99,11 +99,18 @@ maint_off_hs() {
+    docker exec solr cp /hydroshare/schema.xml /opt/solr/server/solr/collection1/conf/schema.xml
+    echo "  - docker exec solr sed -i '/<schemaFactory class=\"ManagedIndexSchemaFactory\">/,+4d' /opt/solr/server/solr/collection1/conf/solrconfig.xml"
+    docker exec solr sed -i '/<schemaFactory class="ManagedIndexSchemaFactory">/,+4d' /opt/solr/server/solr/collection1/conf/solrconfig.xml
+    echo "  - docker exec solr rm /opt/solr/server/solr/collection1/conf/managed-schema"
```
@sblack-usu wondering why this sed call is needed.

sblack-usu replied:
the solr upgrade changes the way schemas are managed

hyi commented on hydroshare/settings.py:
```diff
@@ -290,7 +278,6 @@
     "crispy_forms",
     "mezzanine.accounts",
-    "mezzanine.mobile",
     "haystack",
```
@sblack-usu not sure whether removing "mezzanine.mobile" would cause any issues with styling of HS site on mobile devices, most likely not, but it'd be good if @Maurier can confirm here.

sblack-usu replied:
it was deprecated and has been removed. https://github.com/stephenmcd/mezzanine/commit/f75cd3ce6f58ee5e51e6a02336237381efe1a65e#diff-cb86fa52440c3b29e9ad70ce1a82fe14

sblack-usu commented on hs_app_netCDF/admin.py:
```diff
@@ -1,6 +1,4 @@
 
-# admin.site.register(MyResource, PageAdmin)
-admin.site.register(NetcdfResource, PageAdmin)
+admin.site.unregister(NetcdfResource)
```
I can just remove this file

sblack-usu replied:
I just tested this on dev-hs-1 with an existing database (doing a mock of the deploy from 1.16.1 to 1.17)  The unregister is needed to update the database because the page is registered and stored as registered in the db.  I'll bring back the file and submit a pr

sblack-usu commented:
@hyi, I've addressed all of your comments.  Because of the changes in hs_docker_base, the jenkins build will fail until hs_docker_base rebuilds the django-1.11

sblack-usu commented:
retest this, please
